ui/src/flux/components/TimeMachineTables.tsx

```diff
@@ -73,7 +109,26 @@ class TimeMachineTables extends PureComponent<Props, State> {
   }
 
   private get selectedResult(): FluxTable {
-    return this.props.data.find(d => d.id === this.state.selectedResultID)
+    const selectedTable = this.props.data.find(
+      d => d.id === this.state.selectedResultID
+    )
+    if (selectedTable) {
+      return filterTable(selectedTable)
```

Comment:
Could we move this call out of a getter? Or maybe memoizing it with `memoizeOne` would be the most natural. It's potentially an expensive operation (if `selectedTable.data` is huge), and it would be a shame to execute it more often then we need.

(I ask because I once fixed a performance issue that was eerily similar: https://github.com/influxdata/chronograf/pull/3596/commits/8f74d1f4b4f67f7aa475e7365fdc1ccfd782561b)

Reply:
can do